File: frontend/projects/app/src/app/features/meta/models/subAttachmentDTO.ts
import {MediaDTO} from './mediaDTO';
import {TargetDTO} from './targetDTO';


/**
 * Représente une sous-pièce jointe (par exemple, une photo ou une vidéo dans un album).
 */
export interface SubAttachmentDTO {
  media: MediaDTO;
  target: TargetDTO;
  type: string;
  url: string;
}

File: backend/src/main/java/fr/hoenheimsports/backend/metaservice/dtos/SubAttachmentDTO.java
package fr.hoenheimsports.backend.metaservice.dtos;

import org.jspecify.annotations.Nullable;

/**
 * Represents a sub-attachment (for example, a photo or video inside an album).
 *
 * @param media  The media object of the sub-attachment.
 * @param target The target of the sub-attachment (contains ID and URL).
 * @param type   The type of media (e.g., "photo", "video").
 * @param url    The direct URL to the post of this media on Facebook.
 */
public record SubAttachmentDTO(
        @Nullable MediaDTO media,
        @Nullable TargetDTO target,
        String type,
        @Nullable String url
) {
    public SubAttachmentDTO {
        if (type == null) {
            throw new fr.hoenheimsports.backend.metaservice.exceptions.InvalidMetaDtoException("SubAttachment type cannot be null");
        }
    }
}

